test/datatype/partial.c

```diff
@@ -71,7 +71,7 @@ static void show_neighborhood(double* ptr, int how_many, bool show_hex)
 
 int main( int argc, char* argv[] )
 {
-    opal_datatype_t* vector;
```

Comment:
This should be opal_datatype_t.

Reply:
There's a few places where this variable is used where an `(ompi_datatype_t*)` is expected:

```
1044 partial.c:89:55: warning: passing argument 3 of ‘ompi_datatype_create_contigu\
     ous’ from incompatible pointer type [-Wincompatible-pointer-types]
1045      ompi_datatype_create_contiguous(CONT_COUNT, base, &vector);
1046                                                        ^~~~~~~

1051 partial.c:93:24: warning: passing argument 1 of ‘ompi_datatype_dump’ from inc\
     ompatible pointer type [-Wincompatible-pointer-types]
1052      ompi_datatype_dump(vector);
1053                         ^~~~~~

1058 partial.c:97:31: warning: passing argument 1 of ‘opal_datatype_type_extent’ f\
     rom incompatible pointer type [-Wincompatible-pointer-types]
1059      opal_datatype_type_extent(base, &base_extent);
1060                                ^~~~
```

It seemed better to make it `ompi_datatype_t` and then use the `->super` to get to the `opal_datatype_t` where relevant.

Is that incorrect?

Reply:
The ompi_datatype_t is holding additional information that is unused here. But as we use ompi_datatype functions to manipulate it your fix is correct.